### PR TITLE
BAEL-2724_Closures_in_groovy

### PR DESCRIPTION
**Pull request by SmartyAnsh**

### CODE REVIEW

File: core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        return exchangedAmount
+    }
+    
+    def toggle = { it ->
+        return !it
+    }
+    
+//    def toggle = { value ->
+//        return !value
+//    }
```
Comment (ashleyfrieze) [inline core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy]: Please remove commented out.

File: core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        return result
+    }
+    
+    def volume(Closure area, int... dimensions) {
+        if(dimensions.size() == 3) {
+            
+            //consider dimension[0] = length, dimension[1] = breadth, dimension[2] = height
+            //for cube and cuboid
+            return area(dimensions[0], dimensions[1]) * dimensions[2]
+        } else if(dimensions.size() == 2) {
```
Comment (ashleyfrieze) [inline core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy]: Don't need to use `else` here as the `if` returns. Same below.

File: core-groovy/src/test/groovy/com/baeldung/closures/ClosuresUnitTest.groovy [hunk 3]
```diff
@@ -0,0 +1,110 @@
+package com.baeldung.closures
+
+import spock.lang.Specification
+
+class ClosuresUnitTest extends GroovyTestCase {
+
+    Closures closures = new Closures()
+    
+    void testClosures() {
```
Comment (ashleyfrieze) [inline core-groovy/src/test/groovy/com/baeldung/closures/ClosuresUnitTest.groovy]: Can we have separate tests for each construct with names describing each use case.

File: core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy (hunk 2 above)
Comment (SmartyAnsh) [reply core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy]: please have a close look returns are different for each use case

File: core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy (hunk 1 above)
Comment (SmartyAnsh) [reply core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy]: Done

File: core-groovy/src/test/groovy/com/baeldung/closures/ClosuresUnitTest.groovy (hunk 3 above)
Comment (SmartyAnsh) [reply core-groovy/src/test/groovy/com/baeldung/closures/ClosuresUnitTest.groovy]: Done

File: core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy [hunk 4]
```diff
@@ -0,0 +1,90 @@
+package com.baeldung.closures
+
+//import org.junit.Assert
```
Comment (KevinGilmore) [inline core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy]: Let's avoid commenting out import statements.

Comment (SmartyAnsh) [reply core-groovy/src/main/groovy/com/baeldung/closures/Closures.groovy]: removed commented import statements

**Pull request merged**